src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -169,15 +188,15 @@ private static void AdjustSpans(ArrayBuilder<ClassifiedSpan> spans, TextSpan wid
             // Take all the syntax parts.  However, if any have been overridden by a 
             // semantic part, then choose that one.
 
-            var finalParts = ArrayBuilder<ClassifiedSpan>.GetInstance();
+            using var _ = ArrayBuilder<ClassifiedSpan>.GetInstance(out var finalParts);
             var lastReplacementIndex = 0;
             for (int i = 0, n = syntaxParts.Count; i < n; i++)
             {
                 var syntaxPartAndSpan = syntaxParts[i];
 
                 // See if we can find a semantic part to replace this syntax part.
                 var replacementIndex = semanticParts.FindIndex(
-                    lastReplacementIndex, t => t.TextSpan == syntaxPartAndSpan.TextSpan);
+                    lastReplacementIndex, t => t.TextSpan.OverlapsWith(syntaxPartAndSpan.TextSpan));
```

Comment:
why this change?

Reply:
I believe this is an existing bug in the logic related to the comment I added below:

```
                    // There may be multiple semantic parts corresponding to a single
                    // syntactic part, so we might need to go through a syntactic part
                    // multiple times to verify. For example, this is the case with
                    // verbatim string literals containing string escape characters.
```

Essentially, there's a chance the TextSpan of the semantic part will not exactly match the TextSpan of the syntactic part. This occurs with string literals containing string escape characters. The string escape character will not have the same exact span as the syntactic part and therefore will not be considered. I believe instead calling `OverlapsWith` mitigates the problem.